language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaTreeModifyUtil.java

```diff
@@ -243,32 +242,54 @@ private static boolean importExist(UnusedSymbolsVisitor unusedSymbolsVisitor, AS
                 || unusedSymbolsVisitor.getUnusedImports().containsKey(importValue));
     }
 
-    private static TextEdit constructEdit(
-            UnusedSymbolsVisitor unusedSymbolsVisitor, TextDocument oldTextDocument,
-            ASTModification astModification) {
-        String mapping = BallerinaTreeModifyUtil.resolveMapping(astModification.getType(),
+    private static TextEdit constructEdit(UnusedSymbolsVisitor unusedSymbolsVisitor, TextDocument oldTextDocument,
+                                          ASTModification astModification) {
+
+        String textEdit = BallerinaTreeModifyUtil.resolveMapping(astModification.getType(),
                 astModification.getConfig() == null ? new JsonObject() : astModification.getConfig());
-        if (mapping != null) {
-            boolean doEdit = false;
-            if (DELETE.equals(astModification.getType())) {
-                if (unusedSymbolsVisitor.toBeDeletedRanges().contains(astModification)) {
-                    doEdit = true;
-                }
-            } else {
+        if (textEdit == null) {
+            return null;
+        }
+
+        boolean doEdit = false;
+        if (DELETE.equals(astModification.getType())) {
+            if (unusedSymbolsVisitor.toBeDeletedRanges().contains(astModification)) {
                 doEdit = true;
             }
-            if (doEdit) {
-                LinePosition startLinePos = LinePosition.from(astModification.getStartLine(),
-                        astModification.getStartColumn());
-                LinePosition endLinePos = LinePosition.from(astModification.getEndLine(),
-                        astModification.getEndColumn());
-                int startOffset = oldTextDocument.textPositionFrom(startLinePos);
-                int endOffset = oldTextDocument.textPositionFrom(endLinePos);
-                return TextEdit.from(
-                        TextRange.from(startOffset,
-                                endOffset - startOffset), mapping);
+        } else {
+            doEdit = true;
+        }
+        if (doEdit) {
+            TextRange range = getRange(astModification, oldTextDocument, textEdit);
+            if (range == null) {
+                return null;
             }
+            return TextEdit.from(range, textEdit);
         }
         return null;
     }
+
+    public static TextRange getRange(ASTModification modification, TextDocument oldTextDocument, String textEdit) {
+        LinePosition startLinePos = LinePosition.from(modification.getStartLine(), modification.getStartColumn());
+        LinePosition endLinePos = LinePosition.from(modification.getEndLine(), modification.getEndColumn());
+
+        int startOffset;
+        try {
+            startOffset = oldTextDocument.textPositionFrom(startLinePos);
+            int endOffset = oldTextDocument.textPositionFrom(endLinePos);
```

Comment:
As we As we discussed, we need two try-catch cases:

One for the startPosition (s → eof and diff → 0)

And one for the endPosition (s → s and diff → eof - s)

Reply:
since 2 scenarios throw different exceptions, I guess its okay to add a single try block with multiple catch blocks? its already added in L290

Reply:
anyway didnt implement the fallback logic for case 2 (refer to the TODO) considering it as an edge case and, as we need to test and verify the fallback logic before adding it. WDYT?

Reply:
That's not the case I am mentioning.

There are two lines that can cause the IndexOutOfBounds exception: one for the start and one for the end. In your case, you're assigning EOF to the start even when it's in the correct range.

For example, if EOF = 50 and s = 45:
Your implementation sets s = 50 and diff = 0
Ideally, it should be s = 45 and diff = 5

There should be separate try-catch blocks for each statement, catching the same IndexOutOfBound exception.

Reply:
well that makes sense. even though that case didnt occur in config variable scenarios, it still can happen. 
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/43976/commits/8bf556fd080525c738949db25ee7870591dc431d